build/mergeallmodules.sh

```diff
@@ -0,0 +1,170 @@
+if [ $? != 0 ]; then
+        echo "unzip must be installed"
+        exit 1
+fi
+
+IntermediateDir="../intermediate"
+TargetDir="../target"
+DscModuleIntermediateDir="$IntermediateDir/merging_dsc_modules"
+DscModuleTargetDir="$TargetDir/dsc_modules"
+X64Dir="./x64"
+X86Dir="./x86"
+TestSigningDir=$1
+
+if [ "$TestSigningDir" = "" ]
+then
+	TestSigningDir="../test/config/testsigning"
+fi
+
+if [ -d "$TestSigningDir" ]
+then
+	gpg --list-secret-keys
+	if [ $? != 0 ]
+	then
+		echo "Failed to run the gpg command"
+		exit 1
+	fi
+
+	gpg --list-secret-keys | grep -q '2048R/8C3B51C6'
+	if [ $? != 0 ]
+	then
+		TestGPGKey="$TestSigningDir/gpg.asc"
+		if [ -f "$TestGPGKey" ]
+		then
+			echo "Import the test GPG key"
+			gpg --import $TestGPGKey
+			if [ $? != 0 ]
+			then
+				echo "Failed to import the test GPG key"
+				exit 1
+			fi
+		else
+			echo "Did not find the test GPG key at $TestGPGKey"
+			exit 1
+		fi
+	fi
+
+	SigningKeyFilePath="$TestSigningDir/signingkeys.gpg"
+	SigningKeyPassphrase="$TestSigningDir/passphrase"
+
+	if [ ! -f "$SigningKeyFilePath" ]
+	then
+		echo "GPG key file is not found"
+		exit 1
+	fi
+
+	if [ ! -f "$SigningKeyPassphrase" ]
+	then
+		echo "GPG key passphrase is not found"
+		exit 1
+	fi
+else
+	echo "Signing key directory doesn't exist at location: $TestSigningDir"
+        exit 1
+fi
+
+set -e
+
+if [ ! -d "$IntermediateDir" ]
+then
+	mkdir $IntermediateDir
+fi
+
+if [ ! -d "$TargetDir" ]
+then
+	mkdir $TargetDir
+fi
+
+if [ -d "$DscModuleIntermediateDir" ]
+then
+	rm -rf "$DscModuleIntermediateDir"
+fi
+
+mkdir "$DscModuleIntermediateDir"
+
+for ModuleFilePath in $X64Dir/*.zip;
+do
+	# expected file name: ./x64/nxOMSPlugin_1.0.zip
+	ModuleFileName=`basename $ModuleFilePath`
+	ModuleName=`echo $ModuleFileName | cut -d _ -f 1`
+	
+	if [ -f "$X64Dir/$ModuleFileName" ]
+	then
+		echo "Found x64 module .zip: "$X64Dir/$ModuleFileName""
+	else
+		echo "Did not find x64 module .zip: "$X64Dir/$ModuleFileName""
+		exit 1
+	fi
+	
+	if [ -f "$X86Dir/$ModuleFileName" ]
+	then
+		echo "Found x86 module .zip: "$X86Dir/$ModuleFileName""
+	else
+		echo "Did not find x86 module .zip: "$X86Dir/$ModuleFileName""
+		exit 1
+	fi
+	
+	#Merge modules to ./intermediate/ModuleFileName
+	unzip -o $X86Dir/$ModuleFileName -d ./$DscModuleIntermediateDir
+	unzip -o $X64Dir/$ModuleFileName -d ./$DscModuleIntermediateDir
+	
+	WorkingDir="./$DscModuleIntermediateDir/$ModuleName/"
+	
+	(
+		#Generate Sha256sums
+		echo "Generating sha256sums..."
+		cd $WorkingDir
+		find . -type f -print0 | xargs -0 sha256sum |grep -v sha256sums > ./$ModuleName.sha256sums
+	
+		#Test Sha256sums
+		echo "Test sha256sums..."
+		sha256sum -c $ModuleName.sha256sums
+	)
+
+	#Test sign module
+	echo "Signing .sha256sums file..."
+	cat $SigningKeyPassphrase | gpg --no-default-keyring --keyring $SigningKeyFilePath --yes --passphrase-fd 0 --output ./$WorkingDir/$ModuleName.asc --armor --detach-sign ./$WorkingDir/$ModuleName.sha256sums
+	if [ $? != 0 ]; then
```

Comment:
I'm not 100% sure that form of test will work with `set -e`. Please test this with a bad passphrase or something.

When using `set -e`, we generally use the `if <stmt>` type test, which definitely works. in this case, it would be something like:

```
if ! cat $SigningKeyPassphrase | gpg --no-default-keyring --keyring $SigningKeyFilePath --yes --passphrase-fd 0 --output ./$WorkingDir/$ModuleName.asc --armor --detach-sign ./$WorkingDir/$ModuleName.sha256sums; then
```